src/hooks/useRecommendationsObserver.ts

```diff
@@ -63,6 +66,13 @@ function useRecommendationsObserver(
       });
     };
     // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [menuIsOpen, sections]);
```

Comment:
There was a race condition where `menuIsOpen=true` but the recommendations element was not rendered and the function`trackRecommendationView` was not being called for the first focus but gets fired for subsequent focuses

Adding sections as dependency fixed that but caused multiple `trackRecommendationView` hence the addition if `viewedRecommendations` to keep track of when the recommendation has been viewed as long as the menu is open. Once the menu is closed it restrarts